### DISCUSSION TITLE
Regarding styling

### DISCUSSION THREAD

**Discussion by bq-wrongway** [AllenDang/giu#582] [Q&A]
Hi,
Is there at the moment a way to meaningfully style widgets, and app as whole?
Style section is markes with "to be determined", i assume implementation needs to be decided upon?

And in the meantime is there a way to provide styling for widgets? We can change, font and color as i have seen from some examples.

Thanks

**Comment by gucio321** [comment]
yes, styling is 100% possible, see `examples/setstyle`
if you want to style the whole app, you can also use imgui-native `imgui.PushStyle...`
there is also #533 in progress for CSS styles

**Comment by bq-wrongway** [comment]
Okay, 
i will take a longer look at it. I actually wanted to deal with the Mainwindow.
When creatining main window i would like to change :
Transparency
Border (radius, width)
Disabled top bar ( framless,but still resizible)
But apparently frameless cant be resized, i will keep looking though.

**Reply by gucio321** [reply]
@bqMoonlight so the code, you're looking for is MasterWindowFlags.
```go
wnd := giu.NewMasterWindow("my title", 640,480, giu.MasterWindowFlagsFrameless|giu.MasterWIndowFlagsTransparent)
wnd.SetBgColor(&color.Rgba{}) // this'll set window's background to transparent
```

**Reply by bq-wrongway** [reply]
I used that, issue is that when its frameless i cant resize window on border drag.

**Comment by manuel-koch** [comment]
What do I need to use when I want to change the background color of an image ?
I tried the following, but none seems to result in white background for transparent pixels in the texture:
```
imageBg := color.RGBA{R: 255, G: 255, B: 255, A: 255}
giu.SingleWindow().Layout(
    giu.Row(
	    giu.Style().SetColor(giu.StyleColorPopupBg, imageBg).To(
		    giu.Image(texture).Size(100, 100),
	    ),
	    giu.Style().SetColor(giu.StyleColorChildBg, imageBg).To(
		    giu.Image(texture).Size(100, 100),
	    ),
	    giu.Style().SetColor(giu.StyleColorFrameBg, imageBg).To(
		    giu.Image(texture).Size(100, 100),
	    ),
	    giu.Style().SetColor(giu.StyleColorWindowBg, imageBg).To(
		    giu.Image(texture).Size(100, 100),
	    ),
    ),
)
```
